File: 2502-sort-the-people/2502-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        lst = []
        dct = dict(zip(heights,names))
        #print(dct)
        sortedDict = dict(sorted(dct.items(),reverse=True))
        #print(sortedDict)
        for i in sortedDict:
            lst.append(sortedDict[i])
            #print(lst)
        return lst

File: 2502-sort-the-people/2502-sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        int n = names.length;
        HashMap<Integer,String> map = new HashMap<>();
        for(int i=0;i<n;i++){
            map.put(heights[i],names[i]);
        }
        Arrays.sort(heights);
        String[] ans = new String[n];
        int index = 0;
        for(int i=n-1;i>=0;i--){
            ans[index] = map.get(heights[i]);
            index++;
        }
    return ans;
    }
}